src/unwinder/i386/unwinder_i386.cpp

```diff
@@ -0,0 +1,203 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//
+
+#include "stdafx.h"
+#include "unwinder_i386.h"
+
+#ifdef WIN64EXCEPTIONS
+/*++
+
+Routine Description:
+
+    This function virtually unwinds the specified function by executing its
+    prologue code backward or its epilogue code forward.
+
+    If a context pointers record is specified, then the address where each
+    nonvolatile registers is restored from is recorded in the appropriate
+    element of the context pointers record.
+
+Arguments:
+
+    HandlerType - Supplies the handler type expected for the virtual unwind.
+        This may be either an exception or an unwind handler. A flag may
+        optionally be supplied to avoid epilogue detection if it is known
+        the specified control PC is not located inside a function epilogue.
+
+    ImageBase - Supplies the base address of the image that contains the
+        function being unwound.
+
+    ControlPc - Supplies the address where control left the specified
+        function.
+
+    FunctionEntry - Supplies the address of the function table entry for the
+        specified function.
+
+    ContextRecord - Supplies the address of a context record.
+
+
+    HandlerData - Supplies a pointer to a variable that receives a pointer
+        the the language handler data.
+
+    EstablisherFrame - Supplies a pointer to a variable that receives the
+        the establisher frame pointer value.
+
+    ContextPointers - Supplies an optional pointer to a context pointers
+        record.
+
+    HandlerRoutine - Supplies an optional pointer to a variable that receives
+        the handler routine address.  If control did not leave the specified
+        function in either the prologue or an epilogue and a handler of the
+        proper type is associated with the function, then the address of the
+        language specific exception handler is returned. Otherwise, NULL is
+        returned.
+--*/
+HRESULT
+OOPStackUnwinderX86::VirtualUnwind(
+    __in DWORD HandlerType,
+    __in DWORD ImageBase,
+    __in DWORD ControlPc,
+    __in _PIMAGE_RUNTIME_FUNCTION_ENTRY FunctionEntry,
+    __inout PCONTEXT ContextRecord,
+    __out PVOID *HandlerData,
+    __out PDWORD EstablisherFrame,
+    __inout_opt PKNONVOLATILE_CONTEXT_POINTERS ContextPointers,
+    __deref_opt_out_opt PEXCEPTION_ROUTINE *HandlerRoutine
+    )
+{
+    *EstablisherFrame   = ContextRecord->Esp;
+    *HandlerRoutine     = NULL;
```

Comment:
The HandlerRoutine is an optional pointer, based on the annotation (__deref_opt_out_opt). Can you please wrap the assignment in check it is not NULL?